src/main/java/com/code_intelligence/jazzer/mutation/mutator/lang/FloatingPointMutatorFactory.java

```diff
@@ -0,0 +1,164 @@
+import com.code_intelligence.jazzer.mutation.api.SerializingMutator;
+import com.code_intelligence.jazzer.mutation.mutator.libfuzzer.LibFuzzerMutator;
+import com.google.errorprone.annotations.ForOverride;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.lang.reflect.AnnotatedType;
+import java.lang.reflect.ParameterizedType;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+final class FloatingPointMutatorFactory extends MutatorFactory {
+  @Override
+  public Optional<SerializingMutator<?>> tryCreate(AnnotatedType type, MutatorFactory factory) {
+    if (!(type.getType() instanceof Class)) {
+      return Optional.empty();
+    }
+    Class<?> clazz = (Class<?>) type.getType();
+
+    if (clazz == float.class || clazz == Float.class) {
+      return Optional.of(new AbstractFloatMutator<Float>(type, -Float.MAX_VALUE, Float.MAX_VALUE) {
+        @Override
+        protected Float mutateWithLibFuzzer(Float value) {
+          return LibFuzzerMutator.mutateDefault(value, this, 0);
+        }
+
+        @Override
+        public Float init(PseudoRandom prng) {
+          return (float) initImpl(prng);
+        }
+
+        @Override
+        public Float mutate(Float value, PseudoRandom prng) {
+          return (float) mutateImpl(value, prng);
+        }
+
+        @Override
+        public Float read(DataInputStream in) throws IOException {
+          float value = (float) in.readFloat();
+          return value;
+        }
+
+        @Override
+        public void write(Float value, DataOutputStream out) throws IOException {
+          out.writeFloat(value);
+        }
+      });
+    } else if (clazz == double.class || clazz == Double.class) {
+      return Optional.of(
+          new AbstractDoubleMutator<Double>(type, -Double.MAX_VALUE, Double.MAX_VALUE) {
+            @Override
+            protected Double mutateWithLibFuzzer(Double value) {
+              return LibFuzzerMutator.mutateDefault((Double) value, this, 0);
+            }
+
+            @Override
+            public Double init(PseudoRandom prng) {
+              return (double) initImpl(prng);
+            }
+
+            @Override
+            public Double mutate(Double value, PseudoRandom prng) {
+              value = (double) mutateImpl(value, prng);
+              return value;
+            }
+
+            @Override
+            public Double read(DataInputStream in) throws IOException {
+              double value = (double) in.readDouble();
+              return value;
+            }
+
+            @Override
+            public void write(Double value, DataOutputStream out) throws IOException {
+              out.writeDouble(value);
+            }
+          });
+    } else {
+      return Optional.empty();
+    }
+  }
+
+  // Visible for testing.
+  static abstract class AbstractFloatMutator<T extends Float> extends SerializingMutator<T> {
```

Comment:
Not entirely sure why this class is abstract: If you have one class per type anyway, why not make it not abstract and merge in the implementation of `SerializingMutator` defined anonymously above?

Reply:
I initially thought that the two types can be combined in one. I merged the methods, though still have trouble with types.